# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1263953387 -3600
#      Wed Jan 20 03:09:47 2010 +0100
# Node ID ee18258bc00291bd7268a477e436d788fb2eb736
# Parent  4516a0c97ced3b0037a8403dfa587e49c60ce0e1
Also treat white figure and axes color properties

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,14 @@
 2010-01-20  David Bateman  <dbateman@free.fr>
 
+	* plot/__go_draw_figure__.m: Also plot white foreground and
+	background colors.
+	* plot/plotyy.m: Set axes colors to "none".
+
 	* plot/__go_draw_axes__.m (next_marker):  Remove and all uses of
 	this function.
 	Partially respect the markeredgecolor and markerfacecolor properties.
 	* plot/private/__color__str_rgb__.m: New function
 	* plot/module.mk: Add it here.
 	* plot/__next_line_style__.m: New function selected next line style
 	using the axes linestyleorder property.
 	* plot/__next_line_color.m: Increment next line style here.
diff --git a/scripts/plot/__go_draw_figure__.m b/scripts/plot/__go_draw_figure__.m
--- a/scripts/plot/__go_draw_figure__.m
+++ b/scripts/plot/__go_draw_figure__.m
@@ -65,33 +65,33 @@ function __go_draw_figure__ (h, plot_str
       nkids = length (kids);
 
       if (nkids > 0)
 	fputs (plot_stream, "\nreset;\n");
 	fputs (plot_stream, "set autoscale keepfix;\n");
 	fputs (plot_stream, "set origin 0, 0\n");
 	fputs (plot_stream, "set size 1, 1\n");
         bg = get (h, "color");
-        if (isnumeric (bg) && !isequal (bg, [1,1,1]))
+        if (isnumeric (bg))
           fprintf (plot_stream, "set obj 1 rectangle from screen 0,0 to screen 1,1 behind fc rgb \"#%02x%02x%02x\"\n", 255 * bg);
           bg_is_set = true;
         else
           bg_is_set = false;
         endif
 	for i = 1:nkids
 	  type = get (kids(i), "type");
 	  switch (type)
 	    case "axes"
 	      ## Rely upon listener to convert axes position to "normalized" units.
 	      orig_axes_units = get (kids(i), "units");
 	      orig_axes_position = get (kids(i), "position");
 	      unwind_protect
 		set (kids(i), "units", "normalized");
                 fg = get (kids(i), "color");
-                if (isnumeric (fg) && !isequal (fg, [1,1,1]))
+                if (isnumeric (fg))
                   fprintf (plot_stream, "set obj 2 rectangle from graph 0,0 to graph 1,1 behind fc rgb \"#%02x%02x%02x\"\n", 255 * fg);
                   fg_is_set = true;
                 else
                   fg_is_set = false;
                 endif
 		if (output_to_paper)
 		  axes_position_on_page = orig_axes_position .* paper_position([3, 4, 3 ,4]);
 		  axes_position_on_page(1:2) = axes_position_on_page(1:2) +  paper_position(1:2);
diff --git a/scripts/plot/plotyy.m b/scripts/plot/plotyy.m
--- a/scripts/plot/plotyy.m
+++ b/scripts/plot/plotyy.m
@@ -140,16 +140,17 @@ function [ax, h1, h2] = __plotyy__ (ax, 
   else
     ax(1) = axes ();
   endif
   newplot ();
   h1 = feval (fun1, x1, y1);
 
   set (ax(1), "ycolor", getcolor (h1(1)));
   set (ax(1), "xlim", xlim);
+  set (ax(1), "color", "none");
 
   cf = gcf ();
   set (cf, "nextplot", "add");
 
   if (ishandle(ax(2)) && strcmp (get (ax(2), "type"), "axes"))
     axes (ax(2));
   else
     ax(2) = axes ();
